reactioncommerce/reaction
# Refactor NavBar and PDP to use GraphQL queries.

Resolves
Impact: **major**  
Type: **refactor**

## Issue
Currently the NavBar and PDP gets all its data from Meteor. This is a hurdle for removal of Meteor from the app.

## Solution
- Spilt the components into 2 separate components for customer and admin. In the customer component I am fetching the data from GraphQL queries. 
  - I have changed the code to pass only the required props to child component instead of passing `{...this.props}`, this will prevent a lot of confusion.
  - Nothing has been modified on Admin side.
  - Removed all the admin code from the customer's components.

## Testing

1. Login as admin and create a few tags and sub-tags.
2. Change the order of the tags by dragging them.
3. Create 3 kinds of products.
    - with 1 variant and images
    - with many variants and images
    - with variants and options and images
1. Assign these tags to the products.
4. Check that the tags are correctly ordered.
5. Check that clicking on any tag shows you the respective products.
6. Check that hovering over a tag with sub-tags should display those sub-tags.
1. Change the language and check it works on all pages.
1. Change the currency and check it works on all pages.
1. Clicking on the shop logo should take you to the homepage.
1. Go to PDP and try adding to cart.
1. Add a product to cart and see that cart icon shows 1.
1. Switch between variants and options and check the image is updated.
1. Try adding > 1 quantity of a product and check that the cart is updated.
1. Clicking on tags should take you the tag's page.
1. Clicking on the product from that page should take you to PDP.
1. Add another variant/option and check that cart is updated.
1. Open the cart drawer and check that the total and items are correctly displayed. 
1. Add another item and check that cart drawer is updated.
1. Click on any item and it should take you to that variant/option.
1. Try removing an item from the cart and the total should be updated.
1. Make the cart empty by removing all the item and the message for empty cart should be displayed.
1. Logout and as a anonymous user check the above things.
1. Register as a user and check again the above things.
1. Login as admin and change the product quantity and backorder policy. 
1. Log out and check that correct badges are displayed on the product, also check that you are unable to add a `Sold Out` product to the cart.
